include/counter.h

```diff
@@ -39,6 +39,31 @@ extern "C" {
 
 /**@} */
 
+/**@defgroup COUNTER_TOP_FLAGS Flags used by @ref counter_top_cfg.
+ * @{
+ */
+
+/**
+ * @brief Flag preventing counter reset when top value is changed.
+ *
+ * If flags is set then counter is free running while top value is updated
+ * otherwise counter is reset. If counter is counting up and free running it is
```

Comment:
> If counter is counting up and free running [...]

This is already explained in the description of `counter_set_top_value()` function. Maybe we shouldn't repeat it here?

Reply:
right, i've removed section that overlaps with `counter_set_top_value` description.